Resources/application.py

```diff
@@ -1178,7 +1207,7 @@ def missingDataViz(self):
             return
         dialog = MissingNoGUI.missingDialog(data)
 
-        return        
+        return
 
     def downloadData(self, update):
```

Comment:
I'm not sure this will work as you have written it. Instead of passing the 'por' to the 'DataDownloadV4.py' script we probably need to pass the actual start and end dates. Something like this:

<pre>
if porYes.isChecked():
    endDate = datetime.today() - pd.DateOffset(days=1)
    startDate = lastWaterYear(endDate - pd.DateOffset(years=por)) + pd.DateOffset(days=1)
else:
    startDate = pd.to_datetime(self.dataTab.dataOptions.porT1.text())
    endDate = pd.to_datetime(self.dataTab.dataOptions.porT2.text())
...
DICT = {
    ...
    "START":startDate,
    "END":endDate,
    ...}
</pre>
and then, in DataDownloadV4.py we need to adjust the code to read in the start and end dates from the dictionary and just use them as opposed to calculating them from the POR.

Reply:
I thought about going in that direction but i figured whats the point of specifying an end date that doesn't default to today()-1. Wouldn't this crash the forecasting function if it assumes that it should use most recent data to project forward? With that in mind i figured that we can just keep the current convention of using a POR integer. plus, i didn't want to disrupt too much of the existing DICTs without having a better understanding of them first.

Reply:
Ahh. I see what you're getting at. I didn't notice that you auto-filled the porT2 box and set it to disabled.  I was picturing you wanting to download data from 1981-2010 or something. But you're right: the software would not let you forecast a new value.

Reply:
We may need to further refine the validator on the porT1 box. I'm imagining someone entering in '30' into that box in which case the 'por' would be 1988 years. That input wouldn't be rejected by the software in the line:
<pre>
test=math.sqrt(por)
</pre>
abd would be passed to the datadownloadv4 script. Perhaps add a condition that porP1 must be greater than 1901 or something.

Reply:
~alright... glad to know that i'm at least on the right track... :)

Reply:
will do. added additional validation on T1 >= 1901. I'll commit this once i figure out how to select/highlight duplicate rows on the StationTable